sdk/identity/azure-identity/src/token_credential_impl.cpp

```diff
@@ -152,44 +154,96 @@ AccessToken TokenCredentialImpl::ParseToken(
     std::string const& expiresInPropertyName,
     std::string const& expiresOnPropertyName)
 {
-  try
+  auto const parsedJson = Azure::Core::Json::_internal::json::parse(jsonString);
```

Comment:
Everything from the credentials ultimately gets wrapped into `AuthenticationException` so it would be clear for the user call to `client.DoOperation()` whether the error is with authentication or with the response itself.
We could take the message, but I don't think they are more informative - at least we indicate where the problem is and with what property.

Reply:
> Is it ok for the `nlohmann::json` exception that would be thrown by the call to `json::parse` to be bubbled up to the customer?

I would say no since this is an implementation detail, though I suspect we may be leaking some exceptions in edge cases, which we'd want to wrap. We shouldn't rely on the error messages on a library which we consider implementation and are free to change under the covers.

This is relevant in codegen as well. For example: 
https://github.com/Azure/autorest.cpp/pull/246#issuecomment-1605125152